Docs (Tour Update): Add a GraphQL tutorial to the Dgraph Tour using a Dgraph Cloud backend

verneleem opened the pull request:
Adds a GraphQL tutorial running a Dgraph Cloud backend, along with various enhancements.

Staging link: https://deploy-preview-351--dgraph-tour.netlify.app/tour/

Still to-do:
- [ ] tracking
- [x] widen content pane on tutorial pages, 
- [x] investigate whether we can use a top-and-bottom arrangement for query and results panes (like the Go tutorial)
- [ ] SEO

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/tutorial/351)
<!-- Reviewable:end -->
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
<!-- Dgraph:start -->
Tour Preview: [<img src="https://bl.ocks.org/prashant-shahi/raw/3a9f99bec84231cfe3c0e82cf883f159/0e588d908ad8c8b10958b87ebdd2ba68779ccf4f/dgraph.svg" height="34" align="absmiddle" alt="Dgraph Preview"/>](https://dgraph-tour-3ea46ecbb5-129710.surge.sh)
<!-- Dgraph:end -->

Commit:
add GraphQL intro and basics

The structure and main menus will probably change, just getting the tour content rewritten to start with.

All editor examples are not compatible yet with graphql or graphql schema. WIP

Commit:
fix content types, schema, inputs, and small edits

Commit:
add schema and graphql partials to work with GraphQL schema pushes and queries/mutations.

split vanilla code off from runnable.js into basic.js

add savedinput shortcode to put input form into markdown for endpoint url and api key.

update dgraph-js

add hashes in footer.html to load new files and ignore cache on rebuild

Commit:
add graphql schema lessons patterning off from existing lessons

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dgraph-io/tutorial?pullRequest=351) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dgraph-io/tutorial?pullRequest=351) before we can accept your contribution.<br/><hr/>**verneleem** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dgraph-io/tutorial?pullRequest=351) it.</sub>

Commit:
add capability of partials to show graphql schema with no push and use the graphql playground in a readonly mode

Commit:
add moredata and search content to graphql tour

Commit:
Merge branch 'master' into verneleem/graphql

Commit:
remove js duplicated in webpacked script

Commit:
fix schema window width on wide displays

Commit:
fix runnable type on graphqlschema/10

Commit:
fix graphql variable check for playground and remove extra title from topbar

aaroncarey committed:
Docs (GraphQL Content for Tour): Navigation and edit pass, patch-in to PR #351 (#352)

* Update landing page, edit "Introduction" section

* Fix navigation page

* More edits, main page changes, and spellcheck

Completed edit of /graphqlintro/ and /graphqlbasic/ (1-5), also spellchecked all new files.

Commit:
add tabbed landing page for dql/graphql

Commit:
fix main sidebar menu for dql/graphql

Commit:
add course dotted numbering see graphqlintro/1 for syntax to use

Commit:
add course numbers and course names

Commit:
remove non mapped dql tour pages from graphql tour

Commit:
add course time estimate data

Commit:
show course estimate time to complete

reword landing page

add subheaders to DQL and GraphQL tabs

Commit:
add course completion check on landing page

reroute each course completion back to landing page

Commit:
add element data attribute to trigger write of course completion to localstorage

Reword Dgraph Tour in sidebar menu to Dgraph DQL Tour and Dgraph GraphQL Tour depending on selected course

Commit:
fix bug where schema push without set endpoint opened both the push status and set endpoint modals

Commit:
set hugo version to 0.79.0 necessary for menu .Params

Commit:
fix hugo version to string in netlify.toml

Commit:
fix links

Commit:
fix links

Commit:
fix route first lesson prev button back to landing page

Commit:
add instruction on graphql endpoint change modal with link to Run Dgraph lesson

Commit:
use .Permalink and .Section over .URL

Commit:
fix link to graphql cascade docs instead of dql cascade docs

Commit:
edit wording, since GraphQl tour and DQL tour are side by side instead of continuous, the "later" wording did not make sense

Commit:
chore: clean up this repo.

There was a published directory that does not reflect current published content. This public content is generated in the build script on Netlify when pushed to Github

remove old theme content from previous designs

Commit:
fix 404 page

add redirect for netlify 404 page

Commit:
switch graphql/dql

Commit:
fix module in sidebar menu links to use relative links so they work on preview deploys

Commit:
get live github star data

Commit:
remove live loader, bulk loader, and triples from GraphQL tour content

Commit:
move Data, More Data to the last lesson of the GraphQL tour

aaroncarey committed:
Docs (Editing, writing and testing pass): patch-in to new GraphQL tour content in PR #351 (#353)

* ownsPet -> ownsPets

* More edits

* Edits to "graphqlbasic"

* Update 8.md

* graphqlschema (1.md and 2.md)

* Main page edits

* Completed edit pass

* Complete edit pass and spellcheck

* Incorporate review feedback

bucanero commented:
Staging link: https://deploy-preview-351--dgraph-tour.netlify.app/tour/

aaroncarey committed:
Update 6.md

Removed description of visualization not present in the tour.

aaroncarey committed:
Update 6.md (#354)

MichelDiz commented on content/graphqlintro/1.md:
```diff
@@ -0,0 +1,33 @@
+exclusively on using Dgraph Cloud with GraphQL. {{% /notice %}}
+
+The tutorial is divided into modules. You can access the table of contents at
+any time by clicking "A Tour of Dgraph" in the top left and can move backwards
```
This is confusing. I know it is from the previous text. But we should change it now. The UI doesn't reflect this phrase.

verneleem replied:
Done. removed tour navigation instructions as they are not needed anyways IMO.

MichelDiz commented on content/graphqlmoredata/3.md:
```diff
@@ -0,0 +1,32 @@
++++
+date = "2020-09-15T00:00:00+00:00"
+next = "/?graphql"
+prev = "/graphqlmoredata/3"
```
This step is not showing up. By clicking we are redirected before the page appears.

verneleem replied:
Fixed. This 3rd lesson was added after moving the module. The 2nd lesson now links to the 3rd and the 3rd links back to the 2nd. https://deploy-preview-351--dgraph-tour.netlify.app/tour/graphqlmoredata/3/

Commit:
fix next lesson link

Commit:
fix previous lesson link

Commit:
remove tour navigation direction

The navigation is easy enough to figure out without instructions and the instructions were from a previous version.

Commit:
fixed missing not example

MichelDiz commented:
@verneleem please sign the CLA

bucanero commented on content/basic/10.md:
```diff
@@ -4,7 +4,10 @@ next = "/basic/11"
+course = 2
+courseName = "Basic"
 
 
```
```suggestion

```

bucanero committed:
Update content/basic/10.md

trigger link checker

Commit:
remove static css from compiled javascript file

Commit:
remove vertical runnable on wide displays

fix lesson content panel to make wider on wider displays

Commit:
fix query to actually query the data that the lesson talks about

Commit:
remove ratel from running dgraph commands

Commit:
show ratel in iFrame to match course context about looking at the graph

Commit:
added shortcode to route iFrame with link in lesson

Commit:
feat: make lesson content and runnable resiazable

Commit:
chore: delete committed swap file

Commit:
style resizer bar the Dgraph way
